spec/features/consumer/account_spec.rb

```diff
@@ -60,4 +60,33 @@
     end
   end
 
+  context 'editing user addresses' do
+    before do
+      visit '/account/edit_address'
+    end
+    it 'shows billing address in the form' do
+      expect(page).to have_field('user[bill_address_attributes][firstname]', with: user.bill_address.firstname)
+      expect(page).to have_field('user[bill_address_attributes][address1]', with: user.bill_address.address1)
+      expect(page).to have_field('user[bill_address_attributes][phone]', with: user.bill_address.phone)
+
+      expect(page).to have_field('user[bill_address_attributes][state_id]', with: user.bill_address.state_id)
+      expect(page).to have_field('user[bill_address_attributes][country_id]', with: user.bill_address.country_id)
+    end
+
+    it 'updates shipping address' do
+      fill_in('user[ship_address_attributes][firstname]', with: 'John')
+      fill_in('user[ship_address_attributes][lastname]', with: 'Doe')
+      fill_in('user[ship_address_attributes][address1]', with: '11 Lovely Street')
+      fill_in('user[ship_address_attributes][phone]', with: '123-456-7890')
+      fill_in('user[ship_address_attributes][city]', with: 'Merbourne')
+      fill_in('user[ship_address_attributes][zipcode]', with: '3127')
+
+      click_button('Update')
+
+      expect(page).to have_content '11 Lovely Street'
```

Comment:
Do you need to make sure that the data actually saved?

I'm just imagining an "Update" button that doesn't do anything or works very slowly. The next three lines would just test the currently filled out form that is not submitted yet.